core/cfuns.go

```diff
@@ -14,17 +14,31 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License.
  */
-package client_test
+package rocketmq
 
+/*
+#cgo LDFLAGS: -L/usr/local/lib -lrocketmq
+#include "rocketmq/CMessageExt.h"
+#include "rocketmq/CPushConsumer.h"
+*/
+import "C"
 import (
-    "fmt"
-    "testing"
-    "../client"
+	"sync"
 )
 
-func TestGetMessageTopic(test *testing.T){
-    fmt.Println("-----TestGetMessageTopic Start----")
-    msg := client.CreateMessage("testTopic")
-    client.DestroyMessage(msg)
-    fmt.Println("-----TestGetMessageTopic Finish----")
+var pushConsumerMap sync.Map
+
+//export consumeMessageCallback
+func consumeMessageCallback(cconsumer *C.CPushConsumer, msg *C.CMessageExt) C.int {
+	consumer, exist := pushConsumerMap.Load(cconsumer)
+	if !exist {
+		return C.int(ReConsumeLater)
+	}
+
+	msgExt := cmsgExtToGo(msg)
+	cfunc, exist := consumer.(*defaultPushConsumer).funcsMap.Load(msgExt.Topic)
+	if !exist {
+		return C.int(ReConsumeLater)
```

Comment:
What about add a new error message?

Reply:
it will be implement asap.

Reply:
yes, it should, but the PR was merged, i will add the error in this place or you could open a PR for this :)